packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,753 @@
+  currentPosition = null,
+  currentPositionIcon = <LocationArrow size={13} />,
+  currentPositionUnavailableIcon = <Ban size={13} />,
+  findNearbyStops = () => {},
+  GeocodedOptionIconComponent = GeocodedOptionIcon,
+  geocoderConfig,
+  getCurrentPosition,
+  hideExistingValue = false,
+  initialSearchResults = null,
+  inputPlaceholder = null,
+  isStatic = false,
+  layerColorMap = {},
+  location = null,
+  LocationIconComponent = DefaultLocationIcon,
+  locationType,
+  nearbyStops = [],
+  onLocationSelected,
+  onTextInputClick = null,
+  operatorIconMap = {},
+  preferredLayers = [],
+  sessionOptionIcon = <Search size={13} />,
+  sessionSearches = [],
+  showClearButton = true,
+  showUserSettings = false,
+  sortByDistance = false,
+  stopOptionIcon = <Bus size={13} />,
+  stopsIndex = null,
+  suggestionCount = 3,
+  suppressNearby = false,
+  UserLocationIconComponent = UserLocationIcon,
+  userLocationsAndRecentPlaces = [],
+}: LocationFieldProps): React.ReactElement => {
+  /**
+   * Gets the initial value to place in the input field.
+   */
+  const getValueFromLocation = () => {
+    const label = location?.name || "";
+    return location && !hideExistingValue ? label : "";
+  };
+
+  const listBoxId = `listbox-${optionKey}`;
+
+  const intl = useIntl();
+
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [stateGeocodedFeatures, setGeocodedFeatures] = useState([]);
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [stateMessage, setMessage] = useState(null);
+  const [stateValue, setValue] = useState(getValueFromLocation());
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    // location could be null if none is set
+    setValue(location?.name || "");
+    setGeocodedFeatures([]);
+  }, [location]);
+
+  useEffect(() => {
+    if (initialSearchResults) {
+      setGeocodedFeatures(initialSearchResults);
+      setMenuVisible(true);
+    }
+  }, [initialSearchResults]);
+
+  const geocodeAutocomplete = throttle(1000, text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode autocomplete search.");
+      return;
+    }
+    getGeocoder(geocoderConfig)
+      .autocomplete({ text })
+      // TODO: Better type?
+      .then(
+        (result: {
+          features: any;
+          results: { error: { message: string } };
+        }) => {
+          let message: string;
+          // If no features found in response, default to empty array.
+          let geocodedFeatures = result?.features;
+          if (!geocodedFeatures) {
+            // Get the Pelias error message if exists.
+            // TODO: determine how other geocoders return error messages.
+            const errorMessage = result?.results?.error?.message;
+            // If the result did not contain a list of features, add special note.
+            message = intl.formatMessage({
+              id: "otpUi.LocationField.geocoderUnreachable"
+            }, { error: errorMessage });
```

Comment:
I'd prefer the two-line format as on lines 138-139.

Reply:
Resolved in b4bd8d8c0890c0d8425251b9583773800bc1bc30